Tasks/ServiceFabricUpdateManifests/Update-DockerImageSettings.psm1

```diff
@@ -0,0 +1,61 @@
+function Update-DockerImageSettings
+{
+    [CmdletBinding()]
+    Param()
+
+    Trace-VstsEnteringInvocation $MyInvocation
+    try
+    {
+        $imageDigestsFilePathSearchPattern = Get-VstsInput -Name imageDigestsPath -Require
+        $imageDigestsPath = Find-VstsFiles -LegacyPattern $imageDigestsFilePathSearchPattern
+
+        $appPackagePathSearchPattern = (Get-VstsInput -Name applicationPackagePath -Require)
+        $appPackagePath = Find-VstsFiles -LegacyPattern $appPackagePathSearchPattern -IncludeDirectories
+
+        # Collect the image digest values for the images
+        $imageDigestValues = (Get-Content -Path $imageDigestsPath).Replace("`r`n", "`n").Split("`n")
+        $imageNameToDigestMapping = @{}
+        foreach ($imageDigestValue in $imageDigestValues)
+        {
+            $slashIndex = $imageDigestValue.IndexOf("/")
+            $hashSeparatorIndex = $imageDigestValue.IndexOf("@")
+            if ($slashIndex -lt 0 -or $hashSeparatorIndex -lt 0)
+            {
+                throw (Get-VstsLocString -Key InvalidImageDigestValue -ArgumentList @($imageDigestValue, $imageDigestsPath))
+            }
+            $imageName = $imageDigestValue.Substring($slashIndex + 1, $hashSeparatorIndex - $slashIndex - 1)
+            $imageNameToDigestMapping[$imageName] = $imageDigestValue
+        }
+
+        # Update the service manifests to use the digest-qualified image names
+        $appManifestPath = Join-Path $appPackagePath "ApplicationManifest.xml"
+        $appManifestXml = [xml](Get-Content $appManifestPath)
+        foreach ($serviceManifestImport in $appManifestXml.ApplicationManifest.ServiceManifestImport)
+        {
+            $serviceManifestPath = [System.IO.Path]::Combine($appPackagePath, $serviceManifestImport.ServiceManifestRef.ServiceManifestName, "ServiceManifest.xml")
+            $serviceManifestXml = [xml](Get-Content $serviceManifestPath)
+
+            $hasUpdates = $false
+            $codePackages = @($serviceManifestXml.ServiceManifest.CodePackage)
+            foreach ($codePackage in $codePackages)
+            {
+                if ($codePackage.EntryPoint -and $codePackage.EntryPoint.ContainerHost -and $codePackage.EntryPoint.ContainerHost.ImageName)
+                {
+                    $digest = $imageNameToDigestMapping[$codePackage.EntryPoint.ContainerHost.ImageName]
+                    if ($digest)
+                    {
+                        $codePackage.EntryPoint.ContainerHost.ImageName = $digest
```

Comment:
Don't we need to update service and application version here?

Reply:
No, that's not the point of this update mode.  If you configure the task to use the "Update versions" mode, that will update the service and application versions.  That's handled by the Update-ApplicationVersions.psm1 file.

---
In reply to: [137221269](https://github.com/Microsoft/vsts-tasks/pull/5076#discussion_r137221269) [](ancestors = 137221269)

Reply:
My question was that when we are updating image digest, then we also will need to update version. isn't it? If so, then will user need to add this task twice?